Tests/NIOTests/ByteBufferTest.swift

```diff
@@ -3052,3 +3052,14 @@ extension ByteBufferTest {
     }
     
 }
+
+// MARK: - ExpressibleByArrayLiteral init
+extension ByteBufferTest {
+    
+    func testCreateBufferFromArray() {
+        let bufferView: ByteBufferView = [0x00, 0x01, 0x02]
+        let buffer = ByteBuffer(ByteBufferView(bufferView))
+        
+        XCTAssert(buffer.readableBytesView == [0x00, 0x01, 0x02])
```

Comment:
I'd like to see substantially more testing for equatability here. We should aim to have some tests that prove non-equality works appropriately, that work for slices of BBVs, and so-on.